optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -240,4 +246,153 @@ public void skipAhead() throws ExecutionException, InterruptedException {
         assertTrue(eventCount.get() < 4);
     }
 
+    /**
+     * Tests whether SolverManager can solve on multiple threads problems that use multiple thread counts.
+     */
+    @Test(timeout = 600_000)
+    public void multipleThreadSolvingWithSolverManager() throws ExecutionException, InterruptedException {
+        int processCount = Runtime.getRuntime().availableProcessors();
+        CyclicBarrier barrier = new CyclicBarrier(processCount);
+        final SolverConfig solverConfig = PlannerTestUtils.buildSolverConfig(TestdataSolution.class, TestdataEntity.class)
+                .withPhases(new ConstructionHeuristicPhaseConfig(),
+                            new CustomPhaseConfig().withCustomPhaseCommands(
+                                    scoreDirector -> {
+                                        try {
+                                            barrier.await();
+                                        } catch (InterruptedException | BrokenBarrierException e) {
+                                            fail("Cyclic barrier failed.");
+                                        }
+                                    }))
+                // Adds moveThreadCount to the solver config.
+                .withMoveThreadCount(String.valueOf(processCount));
+        // Creates solverManagerConfig with multiple threads.
+        SolverManager<TestdataSolution, Long> solverManager = SolverManager.create(
+                solverConfig, new SolverManagerConfig().withParallelSolverCount(String.valueOf(processCount)));
+
+        List<SolverJob<TestdataSolution, Long>> jobs = new ArrayList<>();
+        for (long i = 0; i < processCount; i++) {
+            jobs.add(solverManager.solve(i, PlannerTestUtils.generateTestdataSolution("s" + i)));
+        }
+
+        for (SolverJob<TestdataSolution, Long> job : jobs) {
+            assertSolutionInitialized(job.getFinalBestSolution());
+        }
+    }
+
+    /**
+     * Tests whether SolverManager can cope with the situation when more problems than processors are submitted.
+     * In addition, the test verifies whether solving with consumers is affected by this.
+     */
+    @Test(timeout = 600_000)
+    public void moreProblemsThanCpus() throws InterruptedException, ExecutionException {
+        // Use twice the amount of processes than available processors.
+        int problemCount = Runtime.getRuntime().availableProcessors() * 2;
+
+        SolverManager<TestdataSolution, Integer> solverManager = createSolverManagerTestableByDifferentConsumers(problemCount);
```

Comment:
I am not entirely sure I understand what this test aims to do. The `problemCount` != CPU count used by a solver manager.

One can submit 1000 problems while having only 1 CPU used by a solver manager.

Reply:
Yes, the count of problems added (e.g. on line 321) exceeds the cpu count. The idea of the test is to check whether SolverManager can manage to solve more problems than it has cpus. ( https://docs.google.com/document/d/1W_lgu7Aliv1XVX133yUrHG1V7-b97dMNGsbAOc-DGRg/edit# under Testing, the 3rd point )

Reply:
That's fine. By why passing the problemCount to the configuration of SolverManager as the number of threads it should create?

Reply:
Yes, I agree it doesn't make sense, I've removed it.